### ISSUE TITLE
Can you toggle the unit property after init?

### ISSUE THREAD

**Issue by gabaum10** [zhenyanghua/MeasureTool-GoogleMaps-V3#15]
I can't seem to figure out how to do it.  Can you change the unit after the instance has been initialized?  I want to add a toggle in my UI for switching the units on the fly.

**Comment by gabaum10** [comment]
I could work around this shortcoming by destroying the plugin before initializing it with different units... unfortunately, there doesn't seem to be a destroy function either?

**Comment by gabaum10** [comment]
Ok, so I was able to actually do this by basically calling .stop(), then initializing a new measure tool on the map, then calling start again.  Basically like this:

```
              initMeasurementTool () {
                if (this.measureTool) {
                    this.measureTool.end();
                    this.measureTool.removeListener('measure_change');
                    this.measureTool.removeListener('measure_end');
                }
                this.measureTool = new MeasureTool(map, {
                    contextMenu: false,
                    unit: this.units
                });
                this.measureTool.addListener('measure_change', () => {
                    this.hasMeasurements = true;
                });
                this.measureTool.addListener('measure_end', () => {
                    this.hasMeasurements = false;
                });

                if (this.measuring) {
                    this.measureTool.start();
                }
            }
```

**Comment by zhenyanghua** [comment]
Good point, there is actually not setters for the configs at the moment because I almost don't have any time after work to work on the improvements. But hopefully I will be able to get some localization and setters pushed out by the end of this year. 

As for a destroy method, I am following what the convention of Google Maps API is -- the destroy of the instance is implicitly handled by the garbage collector, in javascript, it is automatically disposed when there is no variable points to this instance anymore. So as long as you set the variable that points to this instance to somewhere else or null, it should be destroyed.

**Comment by gabaum10** [comment]
That actually makes sense.  So I guess my work around, while less than ideal should be ok as far as memory leaks go.

Since I'm actually using this for a work project, I'll try to look into the code to see if I can figure out where to set up some setters.  If you point me to the general direction, I can look into it.

**Comment by zhenyanghua** [comment]
To add a setter for `unit` field specifically to the `index.js`, the `helper.js` needs to be modified to add setters for `unit` as well. You can reference what I did inside of the `init()` method in the helper.js to create a setter for the `helper.js`, and then call it from `index.js`. This should be a good start. You might need to extract the switch block out to its own function so you could use it in the setter.

```
  init() {
    switch (this._options.unit.toLowerCase()) {
      case UnitTypeId.METRIC:
        this._lengthMultiplier = 1;
        this.formatLength = this._formatLengthMetric;
        this._areaMultiplier = 1;
        this.formatArea = this._formatAreaMetric;
        break;
      case UnitTypeId.IMPERIAL:
        this._lengthMultiplier = 3.28084;
        this.formatLength = this._formatLengthImperial;
        this._areaMultiplier = 10.7639;
        this.formatArea = this._formatAreaImperial;
        break;
      default:
        this._lengthMultiplier = 1;
        this.formatLength = this._formatLengthMetric;
        this._areaMultiplier = 1;
        this.formatArea = this._formatAreaMetric;
        break;
    }
  }
```

**Comment by zhenyanghua** [comment]
You method should be good as long as your `this.measure` is not reference somewhere else in a `Map`. as long as you keep track of the references and null them when you are done with them there will be no leak.

**Comment by gabaum10** [comment]
@zhenyanghua I have a fix in #16.  If that looks good can we merge it and publish it?

**Comment by zhenyanghua** [comment]
Please see [pull request comment](https://github.com/zhenyanghua/MeasureTool-GoogleMaps-V3/pull/16#issuecomment-344255118) for the next step.
